src/codeCache.h

```diff
@@ -100,13 +100,20 @@ class CodeBlob {
 
 class FrameDesc;
 
+class UnloadProtection;
+
 class CodeCache {
+  friend UnloadProtection;
+
   private:
     char* _name;
+    // E.g. remove (deleted) so that removed library can be reopened
+    char* _clean_name;
```

Comment:
It doesn't need to be stored here - it can be materialized where needed.
Having two name fields is confusing. Furthermore, " (deleted)" is Linux-specific thing.

Reply:
It can be materialized when needed, but then somebody has to free it. If we return a simple `char*` that's going to make things more complicated. How about returning `std::string`?

Reply:
I mean it's better for CodeCache not to deal with `cleanName()`. Having both `name()` and `cleanName()` in the same interface is not really clean (pun intended).
AFAICS, there is only one place in Linux-specific code where you need a name stripped off "(deleted)" suffix, when re-opening a library in UnloadProtection. Why not process the name there? It can be even stored in a stack allocated buffer, so no free is required. Though dynamic allocation is also fine.

Reply:
>  (deleted)" is Linux-specific thing.

This shouldn't be part of `CodeCache`.

Reply:
Yes, it will be helpful to let users know that the orignal file has been deleted and therefore symbol infromation can be incomplete.